### DISCUSSION TITLE
Install subpackages for running tests

### DISCUSSION THREAD

**Discussion by chbndrhnns** [wumpyproject/wumpy#26] [Questions and support] [answered]
I am trying to run all tests in the project from PyCharm. Due to the not-so-common source code layout, I am having trouble to inform PyCharm about the mappings declared in `setup.py` (e.g. `'wumpy.bot': 'library/wumpy-bot/wumpy/bot',`).

Could you guide what's required to run the tests?

**Comment by Bluenix2** [comment, marked as answer]
You can install the library as a PTH-file (similar to a sym-link or the "editable" setup in `setuptools`).

1. Setup a virtual environment and activate it

2. Install `flit` from PyPI with `pip install flit`

3. Navigate into each subpackage and run `flit install --pth-file`

---

Keep in mind that some subpackages depend on others, to ensure that they aren't pulled from PyPI you must install the subpackages in the correct order: `wumpy-gateway`, `wumpy-rest`, `wumpy-models`, `wumpy-cache` and lastly `wumpy-bot`.

Refer to the test workflow inside of `.github/`  to see how it sets up the subpackages for the tests.

**Reply by chbndrhnns** [reply]
Thanks, I learned something new (pth files) today.

**Reply by Bluenix2** [reply]
As an added note, the `setup.py` file is a work-around to allow the library to be installed from the Git repository - it is not the main way of installing the library.

**Reply by chbndrhnns** [reply]
I am seeing an error about a partially imported package when running the tests in the gateway directory. I think l'll start from scratch with your instructions, again.

**Reply by Bluenix2** [reply]
This would not be your fault. I made some changes to `wumpy-rest` which introduced circular imports I had not yet fixed. This should be fixed now.

~~The tests are still failing on commit a84faeeb0aabf585f016cfaf2cb823cff85f7350 because of not having updated `wumpy-testing`.~~ Pushed an update to `wumpy-testing` and re-ran the CI, this now passes.

Sorry you had such a poor first experience

**Reply by chbndrhnns** [reply]
I tried again and the circular imports are gone, thanks!
